Rework `GridFilterPanel` to match the updated Comet CI

fichtnerma opened the pull request:
---
<!-- Everything below this line will be removed from the commit message when the PR is merged -->

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-1030
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>
Single Filter:

![image](https://github.com/user-attachments/assets/3eb0a7a8-f271-4113-a535-125c671cbcb2)
Multiple Filter:

![image](https://github.com/user-attachments/assets/34ce69f2-050c-4986-8afe-f8feca49c470)

</details>

johnnyomair commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -87,18 +88,17 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
             height: "20px",
             marginRight: "10px",
         },
-        panelContent: {
-            [`& .${gridClasses.filterForm}:first-child .${gridClasses.filterFormLinkOperatorInput}`]: {
-                ["@media (max-width: 900px)"]: {
-                    display: "none",
-                },
-            },
-        },
```
Is it really okay to remove this?

fichtnerma replied:
I have changed it to only show the spacing on bigger breakpoints

johnnyomair replied:
But the previous change hid the input altogether. I guess this was intentional on mobile? @jamesricky do you know something about this?

jamesricky replied:
Yes, this is the select to choose between `And`/`Or` which is irrelevant for the first item and therefore hidden by default. Since it's just invisible and still has a height it caused an unintended space above the first select when the mobile styling was applied. 

| Without `display: none` | With `display: none` |
| --- | --- |
| <img width="324" alt="Without display none" src="https://github.com/user-attachments/assets/d406f6eb-a983-4ef2-b136-d159c83b78f7"> | <img width="324" alt="With display none" src="https://github.com/user-attachments/assets/5eeb6205-1d26-47b5-87fe-b2be18c572ae"> |

jamesricky replied:
> So we should keep this?

Yes, otherwise I assume the mobile styling would be broken again

fichtnerma replied:
I have kept the mobile styling just added the required Desktop styling to keep the spacing for the AND/OR field even if only a single row is used 

![image](https://github.com/user-attachments/assets/3eb0a7a8-f271-4113-a535-125c671cbcb2)

johnnyomair commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -87,18 +88,17 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
 
             [`.${inputLabelClasses.root}`]: {
-                display: "none",
+                transform: "translateY(-22px)",
```
This seems odd. Why do we need this?

fichtnerma replied:
This is necessary to display the labels on top of the inputs correctly

johnnyomair commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -155,7 +162,9 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
         },
         paper: {
-            boxShadow: shadows[1],
+            boxShadow: "0px 0px 40px 0px #0000001A",
```
Why isn't it possible to use one of the theme's shadows. @jamesricky should we change that?

jamesricky replied:
See https://github.com/vivid-planet/comet/pull/2457#discussion_r1751351862

johnnyomair commented on .changeset/chatty-spoons-help.md:
```diff
@@ -0,0 +1,5 @@
+"@comet/admin-theme": minor
+---
+
+Customise GridFilterPanel according to UI Requirements
```
```suggestion
Rework `GridFilterPanel` to match the updated Comet CI
```

fichtnerma commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -155,7 +162,9 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
         paper: {
-            boxShadow: shadows[1],
+            boxShadow: "0px 0px 40px 0px #0000001A",
+            border: `1px solid ${palette.grey[100]}`,
```
divider and grey100 have the same hex value but i switched to divider now because it makes more sense

fichtnerma commented:
> The spacings in the screenshot don't seem to match those in the design

I have adjusted the spacings and added new screenshots

jamesricky commented:
> > The spacings in the screenshot don't seem to match those in the design
> 
> I have adjusted the spacings and added new screenshots

Still doesn't seem to match the design perfectly:
- Spacing between the rows, including above the first and below the last
- Spacing between the inputs of a row
- Spacing between the rows delete button and the first input
- Spacing between the input and it's label
- The color of the dividers between the rows
- The icon of the "Add filter" button

<img width="1409" alt="Implementation compared to Figma" src="https://github.com/user-attachments/assets/3f22ef6e-905f-4e5c-92e4-424a18436545">

fichtnerma commented:
> > > The spacings in the screenshot don't seem to match those in the design
> > 
> > 
> > I have adjusted the spacings and added new screenshots
> 
> Still doesn't seem to match the design perfectly:
> 
> * Spacing between the rows, including above the first and below the last
> * Spacing between the inputs of a row
> * Spacing between the rows delete button and the first input
> * Spacing between the input and it's label
> * The color of the dividers between the rows
> * The icon of the "Add filter" button
> 
> <img alt="Implementation compared to Figma" width="1409" src="https://private-user-images.githubusercontent.com/6264317/366760408-3f22ef6e-905f-4e5c-92e4-424a18436545.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMjgwMjksIm5iZiI6MTcyNjEyNzcyOSwicGF0aCI6Ii82MjY0MzE3LzM2Njc2MDQwOC0zZjIyZWY2ZS05MDVmLTRlNWMtOTJlNC00MjRhMTg0MzY1NDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTJUMDc1NTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTg2YTRjZjYxYzJlZGQwNmIxNWFjMTUyYzI5NjVjN2ViODM1N2U2ZGM5OGIwYzc0NjMwYWE5NzkyZTQ3NDAyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XpNYwUw4ubM6_aL6tJAaaEktJBtyByzSjYDCIrnE6JM">

I have adjusted the spacings to match the Figma design.  
But it is currently not possible to change the Icon of the Add Filter Button without updating to a newer MUI Datagrid Version

johnnyomair commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -89,16 +92,37 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
-            padding: spacing(4),
-
+            margin: spacing(5, 4, 0, 4),
+            padding: "10px 5px",
```
```suggestion
            padding: spacing(2, 1),
```

johnnyomair commented:
> But it is currently not possible to change the Icon of the Add Filter Button without updating to a newer MUI Datagrid Version

Could you add a TODO for this?

fichtnerma commented:
> > But it is currently not possible to change the Icon of the Add Filter Button without updating to a newer MUI Datagrid Version
> 
> Could you add a TODO for this?

Done

jamesricky commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -89,16 +93,37 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
         },
         panelContent: {
             [`& .${gridClasses.filterForm}:first-child .${gridClasses.filterFormLinkOperatorInput}`]: {
```
Please avoid using `:first-child` as this causes the following error message in the console: 

```
The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".
```

jamesricky commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
```diff
@@ -89,16 +93,37 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
+                border: "none",
+                marginBottom: spacing(2),
+            },
+            "&:first-child": {
```
Please avoid using `:first-child` as this causes the following error message in the console: 

```
The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".
```

```suggestion
            "&:first-of-type": {
```

jamesricky commented on packages/admin/admin-theme/src/componentsTheme/MuiDataGrid.tsx:
The spacings in the mobile styling seem to be a bit off now...

| Previously | Now |
| --- | --- |
| <img width="320" alt="Previously" src="https://github.com/user-attachments/assets/198c12ed-09a7-4ead-add9-46ff06b3d33b"> | <img width="320" alt="Now" src="https://github.com/user-attachments/assets/a4fd69a0-4a78-4fd9-b8e6-4e9fa78473b8"> |

jamesricky replied:
Was the updated mobile styling actually part of this task? 

<img width="391" alt="New mobile design" src="https://github.com/user-attachments/assets/8f7551c1-7d49-47c4-91c2-ad07963e6864">

fichtnerma replied:
![image](https://github.com/user-attachments/assets/2c39567c-6f39-4cbb-b5ee-4f989ebcfa4e)

I have added the styling for the mobile version or at least the parts that are possible without writing a custom GridFilterPanel

jamesricky replied:
> I have added the styling for the mobile version or at least the parts that are possible without writing a custom GridFilterPanel

What parts exactly are not possible? 

---

A few things I noticed:
- The spacings do not match the design
- The input labels are missing
- At least the word "And" should be fully visible an all devices, there should be enough space available

fichtnerma replied:
- Top Section with close button
- Bottom Section with Apply and Reset filters

fichtnerma replied:
labels and spacing should work now: 
![image](https://github.com/user-attachments/assets/5e599701-fa4d-4f37-b809-19ff9127877a)

johnnyomair reviewed (changes requested):
Please fix the mobile styling.

jamesricky reviewed (changes requested):
Please precisely compare your implementation to the design. 
There are a lot of small inconsistencies.
